src/HeaderCell.ts

```diff
@@ -0,0 +1,60 @@
+import WidgetBase from '@dojo/widget-core/WidgetBase';
+import { RegistryMixin } from '@dojo/widget-core/mixins/Registry';
+import { v } from '@dojo/widget-core/d';
+import { HasColumn, HasSortDetail, HasSortEvent } from './interfaces';
+import { ThemeableMixin, theme, ThemeableProperties } from '@dojo/widget-core/mixins/Themeable';
+
+import * as headerCellClasses from './styles/headerCell.css';
+
+export interface HeaderCellProperties extends ThemeableProperties, HasColumn, HasSortDetail, HasSortEvent { }
+
+export const HeaderCellBase = ThemeableMixin(RegistryMixin(WidgetBase));
+
+@theme(headerCellClasses)
+class HeaderCell extends HeaderCellBase<HeaderCellProperties> {
+	onSortRequest(): void {
+		const {
+			key = '',
+			column,
+			sortDetail,
+			onSortRequest
+		} = this.properties;
+
+		if (onSortRequest && (column.sortable || !column.hasOwnProperty('sortable'))) {
```

Comment:
do you think it would be better to not attach the handler if the column is not sortable?